### PR TITLE
fix(blocks): html clipboard data containing code element should be parsed as html

### PR DESCRIPTION
**Pull request by youhua**
1. If HTMLClipboardData include code，return htmlText2Block
2. remove all of the /g flag, due to 
> JavaScript [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) objects are stateful when they have the [global](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global) or [sticky](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky) flags set (e.g., /foo/g or /foo/y). They store a [lastIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex) from the previous match. Using this internally, test() can be used to iterate over multiple matches in a string of text (with capture groups).
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

### CODE REVIEW

**Comment by zuoxiaodong0815** [comment]
The test case failed. Please take time to look. @youhua

**Comment by youhua** [comment]
The test case work fine in my local env, It looks like a problem caused by multiple instances. How can I try to fix it？

File: packages/blocks/src/__internal__/clipboard/utils/markdown.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+    /(\n\n|^\n?)(( {0,3}([*+-]|\d+[.])[ \t]+)[^\r]+?(¨0|\n{2,}(?=\S)(?![ \t]*(?:[*+-]|\d+[.])[ \t]+)))/m, // list
+    /!\[([^\]]*?)] ?(?:\n *)?\[([\s\S]*?)]()()()()()/, // referenceImage
+    /!\[([^\]]*?)][ \t]*()\([ \t]?<?([\S]+?(?:\([\S]*?\)[\S]*?)?)>?(?: =([*\d]+[A-Za-z%]{0,4})x([*\d]+[A-Za-z%]{0,4}))?[ \t]*(?:(["'])([^"]*?)\6)?[ \t]?\)/, // inlineImage
+    /!\[([^\]]*?)][ \t]*()\([ \t]?<([^>]*)>(?: =([*\d]+[A-Za-z%]{0,4})x([*\d]+[A-Za-z%]{0,4}))?[ \t]*(?:(?:(["'])([^"]*?)\6))?[ \t]?\)/, // crazyImage
+    /!\[([^\]]*?)][ \t]*()\([ \t]?<?(data:.+?\/.+?;base64,[A-Za-z0-9+/=\n]+?)>?(?: =([*\d]+[A-Za-z%]{0,4})x([*\d]+[A-Za-z%]{0,4}))?[ \t]*(?:(["'])([^"]*?)\6)?[ \t]?\)/, // base64Image
     // eslint-disable-next-line
-    /\[((?:\[[^\]]*]|[^\[\]])*)]()[ \t]*\([ \t]?<?([\S]+?(?:\([\S]*?\)[\S]*?)?)>?(?:[ \t]*((["'])([^"]*?)\5))?[ \t]?\)/g, // Generallink
+    /\[((?:\[[^\]]*]|[^\[\]])*)]()[ \t]*\([ \t]?<?([\S]+?(?:\([\S]*?\)[\S]*?)?)>?(?:[ \t]*((["'])([^"]*?)\5))?[ \t]?\)/, // Generallink
     // eslint-disable-next-line
-    /\[((?:\[[^\]]*]|[^\[\]])*)] ?(?:\n *)?\[(.*?)]()()()()/g, // inlineLink
```
Comment (youhua) [inline packages/blocks/src/__internal__/clipboard/utils/markdown.ts:14]: I noticed that there are many empty () brackets here and I am curious about their purpose. Can you please explain why they are here and if they serve any function? Thank you.

Comment (doodlewind) [reply packages/blocks/src/__internal__/clipboard/utils/markdown.ts:14]: /cc @zuoxiaodong0815

Comment (fourdim) [reply packages/blocks/src/__internal__/clipboard/utils/markdown.ts:14]: These brackets captures nothing, we are planning to remove these through refactor this week.
You are welcome to remove these in this pull request.

File: packages/blocks/src/__internal__/clipboard/utils/commons.ts [hunk 2]
```diff
@@ -161,9 +161,10 @@ export async function clipboardData2Blocks(
 
   const textClipData = clipboardData.getData(CLIPBOARD_MIMETYPE.TEXT);
 
+  const isHTMLContianCode = /<code/.test(HTMLClipboardData);
   const shouldConvertMarkdown =
     markdownUtils.checkIfTextContainsMd(textClipData);
-  if (HTMLClipboardData && !shouldConvertMarkdown) {
+  if (HTMLClipboardData && (isHTMLContianCode || !shouldConvertMarkdown)) {
```
Comment (fourdim) [inline packages/blocks/src/__internal__/clipboard/utils/commons.ts:167]: Fix the typo. And make it easier for the reader to understand.

**Review by zuoxiaodong0815 — APPROVED** [review]

**Pull request merged**